### PR TITLE
Add a hold to talk callout component

### PR DESCRIPTION
**Pull request by arzga**
### What

Added a `<HintCallout>`, a non-public component to `<PushToTalkButton>`, which is displayed upon a short, < 350 ms press.

### Why

Enable users to learn how to use the push-to-talk button.

### CODE REVIEW

File: components/src/components/BigTranscript.tsx [hunk 1]
```diff
@@ -61,28 +62,29 @@ export const BigTranscript: React.FC = props => {
...
   const entityProps = useSpring({
-    entityEffect: props.entityType !== null ? 1 : 0,
+    entityEffect: props.word.entityType !== null ? 1 : 0,
     config: { duration: 250 },
   })
 
   return (
-    <TransscriptItemDiv className={props.entityType !== null ? `Entity ${props.entityType}` : ''}>
+    // eslint-disable-next-line
+    <TransscriptItemDiv className={`${props.word.entityType ? 'Entity' : ''} ${props.word.isFinal ? 'Final' : ''} ${props.word.entityType || ''}`}>
```
Comment (radiohead) [inline components/src/components/BigTranscript.tsx]: You can utilise `??` operator to make it a bit cleaner and pass liner checks.
```suggestion
    <TransscriptItemDiv className={`${props.word.entityType ?? 'Entity'} ${props.word.isFinal ? 'Final' : ''} ${props.word.entityType ?? ''}`}>
```

File: components/src/components/Callout.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  backgroundColor = '#000',
+  borderRadius,
+}) => {
+  const springProps = useSpring({
+    v: visible ? 1 : 0,
+    config: { tension: 200 },
+  })
+
+  return (
+    <CalloutContainerDiv ax={sourceAnchors[0]} ay={sourceAnchors[1]} halign={destAnchors[0]} valign={destAnchors[1]} arrowpad={`${arrowSize}${cssUnit}`} onClick={() => onClick()} style={{
```
Comment (radiohead) [inline components/src/components/Callout.tsx]: I suggest using custom type for source / destination anchors. Currently there are no bound checks for the arrays, so it's possible to have undefined behaviour if the user passes zero- or one-size array as `sourceAnchors` / `destAnchors`.

Something like this:
```ts
type AnchorProps = {
  x: string
  y: string
}

type CalloutProps = {
  sourceAnchors?: AnchorProps
  destinationAnchors?: AnchorProps
}
```

**Review by radiohead — CHANGES_REQUESTED** [review]

File: components/src/components/Callout.tsx (hunk 2 above)
Comment (arzga) [reply components/src/components/Callout.tsx]: Fixed

File: components/src/components/BigTranscript.tsx (hunk 1 above)
Comment (arzga) [reply components/src/components/BigTranscript.tsx]: Changed as suggested and removed disabled linting

**Review by radiohead — APPROVED** [review]

**Pull request merged**